src/sergittos/bedwars/listener/GameListener.php

```diff
@@ -403,9 +403,10 @@ public function onPickup(EntityItemPickupEvent $event): void {
             foreach($entity->getInventory()->addItem($event->getItem()) as $remains) {
                 $world->dropItem($origin->getLocation(), $remains, new Vector3(0, 0, 0));
             }
-            if ($entity->getGamemode() === GameMode::SPECTATOR){
-                $event->cancel();
-            }
+        }
+
+        if ($entity instanceof Player and $entity->getGamemode() === GameMode::SPECTATOR){
```

Comment:
That will return error, pmmp uses in gamemode dynamic functions. It should be `GameMode::SPECTATOR()`

Reply:
i don't think this will produce an error, because pmmp does the same thing
https://github.com/pmmp/PocketMine-MP/blob/fb9a74e8799c71ed8292cfa53abe7a4c9204629d/src/player/Player.php#L1199